ex03.exs

```diff
@@ -59,8 +59,38 @@ defmodule Ex03 do
 
   """
 
+  def new_value(list) do
```

Comment:
### -16

I don't see why you are using agents here. They seem to make the code way more complicated.

I was looking for something like:

~~~ elixir
  defp into_chunks(collection, count) do
    chunk_size = div(Enum.count( collection), count )
    Enum.chunk_every( collection, chunk_size)
  end

  def pmap(collection, process_count, function) do
    chunk_mapper = &( Task.async( fn -> Enum.map( &1, function) end ) )

    collection
    |> into_chunks(process_count)
    |> Enum.map( chunk_mapper )
    |> Enum.map( &( Task.await( &1 ) ) )
    |> Enum.concat()
  end
~~~